lang/luajit/Portfile

```diff
@@ -31,31 +34,33 @@ patchfiles          powerpc.patch
 
 post-patch {
     reinplace "s|/usr/local|${prefix}|" ${worksrcpath}/etc/luajit.pc
-}
 
-use_configure       no
+    # Remove -march=i686 from CCOPT_x86
+    reinplace -E {s|-march=[a-z0-9_]+||g} ${worksrcpath}/src/Makefile
+
+    # The build process normally expects git and a .git directory to establish
+    # the version but it includes a fallback for a `.relver` file that we use
+    # here
+    system "echo \"${version_date_s}\" > \"${worksrcpath}/.relver\""
+}
 
 compiler.blacklist  {clang < 700} *gcc-4.2 macports-clang-3.3 macports-clang-3.4
 
 # changes to compiler flags must be made before `CFLAGS=...`
 xcode_workaround.type append_to_compiler_flags
 
-build.target        amalg
-build.args-append   CC="${configure.cc}" \
-                    CFLAGS="${configure.cppflags} ${configure.cflags} [get_canonical_archflags] -DLUAJIT_ENABLE_LUA52COMPAT" \
-                    LDFLAGS="${configure.ldflags} [get_canonical_archflags]" \
-                    PREFIX="${prefix}" \
-                    Q=""
+configure.cflags-append -DLUAJIT_ENABLE_LUA52COMPAT
+makefile.override   CC CFLAGS LDFLAGS PREFIX
 
-destroot.args-append \
-                    PREFIX="${prefix}"
+build.target        amalg
+build.args-append   Q=
 
 post-destroot {
-    ln -s ${prefix}/bin/luajit-${version} ${destroot}${prefix}/bin/luajit
     xinstall -m 755 -d ${destroot}${prefix}/share/doc
     copy ${worksrcpath}/doc/ ${destroot}${prefix}/share/doc/${name}
 }
 
 livecheck.type      regex
-livecheck.url       ${master_sites}.html
-livecheck.regex     {LuaJIT-(\d+(?:\.\d+)*(?:-beta\d+)?).tar.gz}
+livecheck.url       ${github.homepage}/commits/v${branch}.atom
+livecheck.version   ${version_date}
+livecheck.regex     {<updated>([:TZ0-9-]+)</updated>}
```

Comment:
Scanning the feed for this branch we can derive the latest commit's timestamp and compare it to the current one. e.g.

```
luajit seems to have been updated (port version: 2023-11-08T10:05:27Z, new version: 2023-11-09T10:05:27Z)
```

If it's possible, it may be more convenient to somehow process this timestamp such that the output be more like:

```
luajit seems to have been updated (port version: 2.1.1699524327, new version: 2.1.1699624327)
```

but I'm not aware of a way to do that. Any thoughts? It would simplify upgrading this port as the `version` could be updated with the rest of the fields derived from that.

Reply:
No, you would need to modify base to support custom conversion. https://github.com/macports/macports-base/blob/master/src/port1.0/portlivecheck.tcl